### PR TITLE
🌐 Add Chinese translation for `docs/zh/docs/async.md`

### PR DESCRIPTION
**Pull request by mkdir700**
This PR translates async.md in Chinese.

### CODE REVIEW

File: docs/zh/docs/async.md [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+
+但在这种情况下，如果你能带上 8 名前收银员/厨师，现在是清洁工一起清扫，他们中的每一个人（加上你）都能占据房子的一个区域来清扫，你就可以在额外的帮助下并行的更快地完成所有工作。
+
+在这个场景中，每个清洁工（包括您）都将是一个处理器，完成这个工作的一部分。
+
+由于大多数执行时间是由实际工作（而不是等待）占用的，并且计算机中的工作是由 <abbr title="Central Processing Unit">CPU</abbr> 完成的，所以他们称这些问题为"CPU 绑定"。
+
+---
+
+CPU 绑定操作的常见示例是需要复杂的数学处理。
```
Comment (Jedore) [inline docs/zh/docs/async.md:277]:
```suggestion
CPU 密集型操作的常见示例是需要复杂的数学处理。
```

File: docs/zh/docs/async.md [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+
+你可以像汉堡的例子那样轮流执行，先是客厅，然后是厨房，但因为你不需要等待 🕙 ，对于任何事情都是清洁，清洁，还是清洁，不会影响任何事情。
+
+无论是否轮流执行（并发），都需要相同的时间来完成，而你也会完成相同的工作量。
+
+但在这种情况下，如果你能带上 8 名前收银员/厨师，现在是清洁工一起清扫，他们中的每一个人（加上你）都能占据房子的一个区域来清扫，你就可以在额外的帮助下并行的更快地完成所有工作。
+
+在这个场景中，每个清洁工（包括您）都将是一个处理器，完成这个工作的一部分。
+
+由于大多数执行时间是由实际工作（而不是等待）占用的，并且计算机中的工作是由 <abbr title="Central Processing Unit">CPU</abbr> 完成的，所以他们称这些问题为"CPU 绑定"。
```
Comment (Jedore) [inline docs/zh/docs/async.md:273]:
```suggestion
由于大多数执行时间是由实际工作（而不是等待）占用的，并且计算机中的工作是由 <abbr title="Central Processing Unit">CPU</abbr> 完成的，所以他们称这些问题为"CPU 密集型"。
```

File: docs/zh/docs/async.md [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+* **音频**或**图像**处理；
+* **计算机视觉**: 一幅图像由数百万像素组成，每个像素有3种颜色值，处理通常需要同时对这些像素进行计算；
+* **机器学习**: 它通常需要大量的"矩阵"和"向量"乘法。想象一个包含数字的巨大电子表格，并同时将所有数字相乘；
+* **深度学习**:这是机器学习的一个子领域，同样适用。只是没有一个数字的电子表格可以相乘，而是一个庞大的数字集合，在很多情况下，你需要使用一个特殊的处理器来构建和使用这些模型。
+
+### 并发 + 并行: Web + 机器学习
+
+使用 **FastAPI**，您可以利用 Web 开发中常见的并发机制（NodeJS 的主要吸引力）。
+
+并且，您也可以利用并行和多进程（让多个进程并行运行）的优点来处理与机器学习系统中类似 **CPU 绑定** 的工作。
```
Comment (Jedore) [inline docs/zh/docs/async.md:290]:
```suggestion
并且，您也可以利用并行和多进程（让多个进程并行运行）的优点来处理与机器学习系统中类似 **CPU 密集型** 的工作。
```

File: docs/zh/docs/async.md [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+* 来自客户端的数据将通过网络发送
+* 客户端接收来自网络中的数据
+* 磁盘中要由系统读取并提供给程序的文件的内容
+* 程序提供给系统的要写入磁盘的内容
+* 一个 API 的远程调用
+* 一个数据库操作，直到完成
+* 一个数据库查询，直到返回结果
+* 等等.
+
+这个执行的时间大多是在等待 <abbr title="Input and Output">I/O</abbr> 操作，因此它们被叫做 "I/O 绑定" 操作。
```
Comment (Jedore) [inline docs/zh/docs/async.md:85]:
```suggestion
这个执行的时间大多是在等待 <abbr title="Input and Output">I/O</abbr> 操作，因此它们被叫做 "I/O 密集型" 操作。
```

File: docs/zh/docs/async.md [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+> Python 的现代版本可以通过使用 `async` 和 `await` 语法创建**协程**，并用于支持**异步代码**。
+
+现在应该能明白其含义了。✨
+
+所有这些都是 FastAPI（通过 Starlette）的动力，也是它拥有如此令人印象深刻的性能的原因。
+
+## 非常技术性的细节
+
+!!! warning
+你可以跳过这里。
```
Comment (Jedore) [inline docs/zh/docs/async.md:398]:
```suggestion
    你可以跳过这里。
```

File: docs/zh/docs/async.md [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+
+    这些都是 FastAPI 如何在内部工作的技术细节。
+
+    如果您有相当多的技术知识（协程、线程、阻塞等），并且对 FastAPI 如何处理 `async def` 与常规 `def` 感到好奇，请继续。
+
+### 路径操作函数
+
+当你使用 `def` 而不是 `async def` 来声明一个*路径操作函数*时，它运行在外部的线程池中并等待其结果，而不是直接调用（因为它会阻塞服务器）。
+
+如果您使用过另一个不以上述方式工作的异步框架，并且您习惯于用普通的 `def` 定义普通的仅计算路径操作函数，以获得微小的性能增益（大约100纳秒），请注意，在 FastAPI 中，效果将完全相反。在这些情况下，最好使用 `async def`，除非路径操作函数内使用执行阻塞的代码。
```
Comment (Jedore) [inline docs/zh/docs/async.md:408]:
```suggestion
如果您使用过另一个不以上述方式工作的异步框架，并且您习惯于用普通的 `def` 定义普通的仅计算路径操作函数，以获得微小的性能增益（大约100纳秒），请注意，在 FastAPI 中，效果将完全相反。在这些情况下，最好使用 `async def`，除非路径操作函数内使用执行阻塞 <abbr title="输入/输出：磁盘读写，网络通讯.">I/O</abbr> 的代码。
```

File: docs/zh/docs/async.md [hunk 7]
```diff
@@ -0,0 +1,432 @@
...
+
+这与 FastAPI 为您调用*路径操作函数*和依赖项的逻辑相反。
+
+如果你的函数是通过 `def` 声明的，它将被直接调用（在代码中编写的地方），而不会在线程池中，如果这个函数通过 `async def` 声明，当在代码中调用时，你就应该使用 `await` 等待函数的结果。
+
+---
+
+再次提醒，这些是非常技术性的细节，如果你来搜索它可能对你有用。
+
+否则，您最好应该遵守的指导原则<a href="#in-a-hurry">赶时间吗？</a>.
```
Comment (Jedore) [inline docs/zh/docs/async.md:432]:
```suggestion
否则，您最好应该遵守的指导原则<a href="#_1">赶时间吗？</a>.
```

**Review by Jedore — CHANGES_REQUESTED** [review]
Some suggestion changes, especially for "CPU bound"/"IO bound"

**Comment by mkdir700** [comment]
thank you @Jedore

**Review by Jedore — APPROVED** [review]
well done 👍

File: docs/zh/docs/async.md (hunk 1 above)
Comment (Jedore) [reply docs/zh/docs/async.md:275]: @mkdir700  what abous this?

File: docs/zh/docs/async.md (hunk 2 above)
Comment (mkdir700) [reply docs/zh/docs/async.md:271]: I'm sorry.I didn't see it.

**Comment by Lenclove** [comment]
Well, it seems someone have checked yet. Do I need to check the translation or give some suggestion? It seems this translation is done.

File: docs/zh/docs/async.md [hunk 8]
```diff
@@ -0,0 +1,430 @@
...
+例如：
+
+* **音频**或**图像**处理；
+* **计算机视觉**: 一幅图像由数百万像素组成，每个像素有3种颜色值，处理通常需要同时对这些像素进行计算；
+* **机器学习**: 它通常需要大量的"矩阵"和"向量"乘法。想象一个包含数字的巨大电子表格，并同时将所有数字相乘；
+* **深度学习**:这是机器学习的一个子领域，同样适用。只是没有一个数字的电子表格可以相乘，而是一个庞大的数字集合，在很多情况下，你需要使用一个特殊的处理器来构建和使用这些模型。
+
+### 并发 + 并行: Web + 机器学习
+
+使用 **FastAPI**，您可以利用 Web 开发中常见的并发机制（NodeJS 的主要吸引力）。
```
Comment (Lenclove) [inline docs/zh/docs/async.md:286]: 使用 **FastAPI**，您可以利用 Web 开发中常见的并发机制的优势（NodeJS 的主要吸引力）。

File: docs/zh/docs/async.md [hunk 9]
```diff
@@ -0,0 +1,430 @@
...
+
+## 结论
+
+让我们再来回顾下上文所说的：
+
+> Python 的现代版本可以通过使用 `async` 和 `await` 语法创建**协程**，并用于支持**异步代码**。
+
+现在应该能明白其含义了。✨
+
+所有这些都是 FastAPI（通过 Starlette）的动力，也是它拥有如此令人印象深刻的性能的原因。
```
Comment (Lenclove) [inline docs/zh/docs/async.md:391]: 所有这些使得FastAPI（通过 Starlette）如此强大，也是它拥有如此令人印象深刻的性能的原因。

File: docs/zh/docs/async.md [hunk 10]
```diff
@@ -0,0 +1,430 @@
...
+
+    这些都是 FastAPI 如何在内部工作的技术细节。
+
+    如果您有相当多的技术知识（协程、线程、阻塞等），并且对 FastAPI 如何处理 `async def` 与常规 `def` 感到好奇，请继续。
+
+### 路径操作函数
+
+当你使用 `def` 而不是 `async def` 来声明一个*路径操作函数*时，它运行在外部的线程池中并等待其结果，而不是直接调用（因为它会阻塞服务器）。
+
+如果您使用过另一个不以上述方式工作的异步框架，并且您习惯于用普通的 `def` 定义普通的仅计算路径操作函数，以获得微小的性能增益（大约100纳秒），请注意，在 FastAPI 中，效果将完全相反。在这些情况下，最好使用 `async def`，除非路径操作函数内使用执行阻塞 <abbr title="输入/输出：磁盘读写，网络通讯.">I/O</abbr> 的代码。
```
Comment (Lenclove) [inline docs/zh/docs/async.md:406]: 如果您使用过另一个不以上述方式工作的异步框架，并且您习惯于用普通的 `def` 定义普通的仅计算路径操作函数，以获得微小的性能增益（大约100纳秒），请注意，在 FastAPI 中，效果将完全相反。在这些情况下，最好使用 `async def`，除非*路径操作函数*内使用执行阻塞 <abbr title="输入/输出：磁盘读写，网络通讯.">I/O</abbr> 的代码。

**Review by Lenclove — COMMENTED** [review]
Hi, I am very glad to review these translation. Could you please give me some advice for these changes?

**Comment by mkdir700** [comment]
> 
👍  very nice, you can try use this:

File: docs/zh/docs/async.md (hunk 10 above)
Comment (mkdir700) [reply docs/zh/docs/async.md:406]: Sorry bro, I didn't find the difference between the two.

`<abbr title="输入/输出：磁盘读写，网络通讯.">I/O</abbr>` ，This sentence is needed.

image

**Review by Lenclove — CHANGES_REQUESTED** [review]
Improve Chinese translation for async.md

**Comment by mkdir700** [comment]
> 

Or give suggestions instead of comments.

image

**Comment by Jedore** [comment]
@Lenclove  You could try following steps, then suggested changes will like https://github.com/tiangolo/fastapi/pull/5591#discussion_r1016301828
image
image

**Comment by Lenclove** [comment]
> @Lenclove You could try following steps, then suggested changes will like [#5591 (comment)](https://github.com/tiangolo/fastapi/pull/5591#discussion_r1016301828) image image

Yes. It's awesome! I have experienced these operations which are very good.

**Review by Lenclove — APPROVED** [review]
Hi, I think it's OK to merge.

**Comment by Lenclove** [comment]
It seems that `coverage` build steps interrupt this pull request. Could we solve it?

**Comment by tiangolo** [comment]
📝 Docs preview for commit 1ad2e171f6d3cc19294772ce2a6bad76b158571b at: https://649a16c47e9b761214ba4bcd--fastapi.netlify.app

**Comment by wdh99** [comment]
`docs/zh/docs/async.md` 文件已经不存在了

**Review by wdh99 — CHANGES_REQUESTED** [review]

**Review by xzmeng — APPROVED** [review]
LGTM

**Comment by tiangolo** [comment]
📝 Docs preview for commit ce7c520f356e3e7bf87f3007c2aa15e513031da6 at: https://c3d2ae96.fastapitiangolo.pages.dev

**Comment by tiangolo** [comment]
Thank you @mkdir700! 🙇 

And thanks for the reviews @Jedore, @Lenclove, @xzmeng, @wdh99 🍰

**Pull request merged**